### PR TITLE
Feat/pool client api module

### PR DESCRIPTION
**Pull request by fabiodmota**
Add maven module pool client and moving it to the pool module

### CODE REVIEW

File: bpdm-pool/src/test/kotlin/org/eclipse/tractusx/bpdm/pool/controller/AddressControllerIT.kt [hunk 1]
```diff
@@ -390,15 +398,19 @@ class AddressControllerIT @Autowired constructor(
             RequestValues.addressPartnerUpdate2.copy(bpn = "BPNLXXXXXXXX"),
             RequestValues.addressPartnerUpdate3.copy(bpn = "BPNAXXXXXXXX")
         )
-        val response = webTestClient.invokePutWithArrayResponse<AddressPartnerResponse>(EndpointValues.CATENA_ADDRESSES_PATH, toUpdate)
+
+
+        val response = poolClient.addresses().updateAddresses(toUpdate);
```
Comment (nicoprow) [inline bpdm-pool/src/test/kotlin/org/eclipse/tractusx/bpdm/pool/controller/AddressControllerIT.kt:403]: Unnecessary semicolon

File: bpdm-pool/src/test/kotlin/org/eclipse/tractusx/bpdm/pool/controller/LegalEntityControllerIT.kt [hunk 2]
```diff
@@ -265,10 +258,9 @@ class LegalEntityControllerIT @Autowired constructor(
         }.first() // search for first
 
         val identifierToFind = expected.properties.identifiers.first()
-        val response = webTestClient.invokeGetEndpoint<LegalEntityPartnerResponse>(
-            "${EndpointValues.CATENA_LEGAL_ENTITY_PATH}/${identifierToFind.value}",
-            "idType" to identifierToFind.type.technicalKey
-        )
+
+
+       val response = poolClient.legalEntities().getLegalEntity(identifierToFind.value,identifierToFind.type.technicalKey);
```
Comment (nicoprow) [inline bpdm-pool/src/test/kotlin/org/eclipse/tractusx/bpdm/pool/controller/LegalEntityControllerIT.kt:263]: Unnecessary semicolon

File: bpdm-pool/src/test/kotlin/org/eclipse/tractusx/bpdm/pool/controller/LegalEntityControllerIT.kt [hunk 3]
```diff
@@ -300,10 +292,9 @@ class LegalEntityControllerIT @Autowired constructor(
 
         var identifierToFind = expected.properties.identifiers.first()
         identifierToFind = identifierToFind.copy(value = changeCase(identifierToFind.value))
-        val response = webTestClient.invokeGetEndpoint<LegalEntityPartnerResponse>(
-            "${EndpointValues.CATENA_LEGAL_ENTITY_PATH}/${identifierToFind.value}",
-            "idType" to identifierToFind.type.technicalKey
-        )
+
+
+        val response = poolClient.legalEntities().getLegalEntity(identifierToFind.value,identifierToFind.type.technicalKey);
```
Comment (nicoprow) [inline bpdm-pool/src/test/kotlin/org/eclipse/tractusx/bpdm/pool/controller/LegalEntityControllerIT.kt:297]: Unnecessary semicolon

File: bpdm-pool/src/test/kotlin/org/eclipse/tractusx/bpdm/pool/controller/SiteControllerIT.kt [hunk 4]
```diff
@@ -247,8 +233,8 @@ class SiteControllerIT @Autowired constructor(
             RequestValues.siteUpdate2.copy(bpn = bpnS1),
             RequestValues.siteUpdate3.copy(bpn = bpnS2)
         )
-        val response = webTestClient.invokePutWithArrayResponse<SitePartnerCreateResponse>(EndpointValues.CATENA_SITES_PATH, toUpdate)
 
+        val response = poolClient.sites().updateSite(toUpdate);
```
Comment (nicoprow) [inline bpdm-pool/src/test/kotlin/org/eclipse/tractusx/bpdm/pool/controller/SiteControllerIT.kt:237]: Unnecessary semicolon

File: bpdm-pool/src/test/kotlin/org/eclipse/tractusx/bpdm/pool/controller/SiteControllerIT.kt [hunk 5]
```diff
@@ -351,9 +333,9 @@ class SiteControllerIT @Autowired constructor(
         testHelpers.assertRecursively(actuals).ignoringFields(SitePartnerCreateResponse::bpn.name).isEqualTo(expected)
     }
 
-    private fun requestSite(bpnSite: String) =
-        webTestClient.invokeGetEndpoint<SitePartnerSearchResponse>(EndpointValues.CATENA_SITES_PATH + "/${bpnSite}")
+    private fun requestSite(bpnSite: String) = poolClient.sites().getSite(bpnSite)
+
+
+    private fun requestSitesOfLegalEntity(bpn: String) = poolClient.legalEntities().getSites(bpn,PaginationRequest());
```
Comment (nicoprow) [inline bpdm-pool/src/test/kotlin/org/eclipse/tractusx/bpdm/pool/controller/SiteControllerIT.kt:339]: Unnecessary semicolon

**Review by nicoprow — CHANGES_REQUESTED** [review]
The package structure is a bit unconventional for a library. My suggestion would be to rename the packages to:
config -> client
dto -> model (as in model of the api)
service -> (extract api interfaces to root, so the package name is bpdm.pool.api)

In the pool test directory is the util.EndpointValues class which now contains some unused constants since you have replaced the webclient with the poolclient. Please remove the unused constants there.

**Comment by fabiodmota** [comment]
All changed according to the requests

**Review by nicoprow — APPROVED** [review]

**Pull request merged**